examples/verify_stone_proof.rs

```diff
@@ -0,0 +1,117 @@
+use ethers::{
+    core::k256::ecdsa::SigningKey,
+    middleware::SignerMiddleware,
+    providers::{Http, Middleware, Provider},
+    signers::{LocalWallet, Signer},
+    types::{Address, U256, U64},
+    utils::{hex, Anvil},
+};
+use stark_evm_adapter::{
+    annotated_proof::AnnotatedProof,
+    annotation_parser::{split_fri_merkle_statements, SplitProofs},
+    ContractFunctionCall,
+};
+use std::{convert::TryFrom, env, str::FromStr, sync::Arc};
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // setup a local mainnet fork
+    let url = env::var("MAINNET_RPC");
+    let forked_url = env::var("FORKED_MAINNET_RPC");
+    // check either env MAINNET_RPC or FORK_MAINNET_RPC is set
+    if url.is_err() && forked_url.is_err() {
+        panic!(
+            "Either MAINNET_RPC or FORK_MAINNET_RPC must be set in env. \
+        You can get a mainnet RPC url from https://infura.io/, \
+        or forked mainnet RPC url from https://tenderly.co/"
+        );
+    }
```

Comment:
Different from the mainnet, the forked is the one for local testing.

In the demo script, when a vanilla mainnet RPC is provided, it will use the anvil local node to fork the mainnet. So the demo script can run test on a forked node even if you provided the vanilla mainnet RPC.

On the other hand, if forked mainnet RPC is provided, it won't need to use the anvil to fork it.